Adding machine set to target cluster induces go panic

1. Bootstrap a cluster
2. Add a machine set of nodes. (either to target cluster or bootstrap cluster.
3. ???
4. Go Panic.

**Machineset.yaml**
```
apiVersion: cluster.k8s.io/v1alpha1
kind: MachineSet
metadata:
  name: test-machineset-1
spec:
  replicas: 2
  selector:
    matchLabels:
      node-type: worker-node
  template:
      metadata:
        labels:
          node-type: worker-node
  providerSpec:
    apiVersion: "vsphereproviderconfig/v1alpha1"
    kind: "VsphereMachineProviderConfig"
    machineSpec:
      machineSpec:
        datacenter: "my-datacenter"
        datastore: "my-datastore"
        resourcePool: "my-resource-pool"
        networks:
        - networkName: "my-network"
          ipConfig:
            networkType: dhcp
        numCPUs: 2
        memoryMB: 2048
        template: "ubuntu-template"
    versions:
      kubelet: 1.11.3
    roles:
    - Node

```

**Output**
```
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
        panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0xedbd96]

goroutine 149 [running]:
sigs.k8s.io/cluster-api-provider-vsphere/vendor/k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
        /go/src/sigs.k8s.io/cluster-api-provider-vsphere/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:58 +0x107
panic(0x14cc1e0, 0x25a28f0)
        /usr/local/go/src/runtime/panic.go:502 +0x229
sigs.k8s.io/cluster-api-provider-vsphere/pkg/cloud/vsphere/utils.GetMachineProviderSpec(0x0, 0x0, 0xc42002bb00, 0x0, 0xc4207b5830)
        /go/src/sigs.k8s.io/cluster-api-provider-vsphere/pkg/cloud/vsphere/utils/utils.go:86 +0x46
sigs.k8s.io/cluster-api-provider-vsphere/pkg/cloud/vsphere/utils.GetVMId(0xc4207f8580, 0xc4200c9ea0, 0x190b560, 0xc42047f640, 0xc4200c9ea0)
        /go/src/sigs.k8s.io/cluster-api-provider-vsphere/pkg/cloud/vsphere/utils/utils.go:112 +0x42
sigs.k8s.io/cluster-api-provider-vsphere/pkg/cloud/vsphere/provisioner/govmomi.(*Provisioner).Exists(0xc42073a4b0, 0x190b5a0, 0xc42009e038, 0xc42080c000, 0xc4207f8580, 0x0, 0x0, 0x0)
        /go/src/sigs.k8s.io/cluster-api-provider-vsphere/pkg/cloud/vsphere/provisioner/govmomi/exists.go:23 +0xf4
sigs.k8s.io/cluster-api-provider-vsphere/pkg/cloud/vsphere.(*VsphereClient).Exists(0xc420739440, 0x190b5a0, 0xc42009e038, 0xc42080c000, 0xc4207f8580, 0x0, 0x0, 0x18fb9e0)
        /go/src/sigs.k8s.io/cluster-api-provider-vsphere/pkg/cloud/vsphere/machineactuator.go:94 +0x65
sigs.k8s.io/cluster-api-provider-vsphere/vendor/sigs.k8s.io/cluster-api/pkg/controller/machine.(*ReconcileMachine).Reconcile(0xc420726700, 0xc420042ae9, 0x7, 0xc42003ee80, 0x11, 0x4, 0x0, 0x0, 0x0)
        /go/src/sigs.k8s.io/cluster-api-provider-vsphere/vendor/sigs.k8s.io/cluster-api/pkg/controller/machine/controller.go:158 +0x3d9
sigs.k8s.io/cluster-api-provider-vsphere/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc4206861e0, 0x0)
        /go/src/sigs.k8s.io/cluster-api-provider-vsphere/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:207 +0x139
sigs.k8s.io/cluster-api-provider-vsphere/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func1()
        /go/src/sigs.k8s.io/cluster-api-provider-vsphere/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:157 +0x36
sigs.k8s.io/cluster-api-provider-vsphere/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1(0xc420390730)
        /go/src/sigs.k8s.io/cluster-api-provider-vsphere/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133 +0x54
sigs.k8s.io/cluster-api-provider-vsphere/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc420390730, 0x3b9aca00, 0x0, 0x20001, 0xc4204840c0)
        /go/src/sigs.k8s.io/cluster-api-provider-vsphere/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:134 +0xbd
sigs.k8s.io/cluster-api-provider-vsphere/vendor/k8s.io/apimachinery/pkg/util/wait.Until(0xc420390730, 0x3b9aca00, 0xc4204840c0)
        /go/src/sigs.k8s.io/cluster-api-provider-vsphere/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88 +0x4d
created by sigs.k8s.io/cluster-api-provider-vsphere/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start
        /go/src/sigs.k8s.io/cluster-api-provider-vsphere/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:156 +0x388
```

--- a/pkg/cloud/vsphere/utils/utils.go
+++ b/pkg/cloud/vsphere/utils/utils.go
@@ -83,6 +83,10 @@ func GetClusterProviderStatus(cluster *clusterv1.Cluster) (*vsphereconfigv1.Vsph
 func GetMachineProviderSpec(providerSpec clusterv1.ProviderSpec) (*vsphereconfigv1.VsphereMachineProviderConfig, error) {
 	config := &vsphereconfigv1.VsphereMachineProviderConfig{}
 
+	if providerSpec.Value == nil {
+		return nil, fmt.Errorf("machine providerconfig is invalid (nil)")
+	}
+
 	err := yaml.Unmarshal(providerSpec.Value.Raw, config)
 	if err != nil {
 		return nil, fmt.Errorf("machine providerconfig unmarshalling failure: %s", err.Error())
@@ -92,6 +96,10 @@ func GetMachineProviderSpec(providerSpec clusterv1.ProviderSpec) (*vsphereconfig
 
 func GetClusterProviderSpec(providerSpec clusterv1.ProviderSpec) (*vsphereconfigv1.VsphereClusterProviderConfig, error) {
 	config := &vsphereconfigv1.VsphereClusterProviderConfig{}
+
+	if providerSpec.Value == nil {
+		return nil, fmt.Errorf("cluster providerconfig is invalid (nil)")
+	}
 
 	err := yaml.Unmarshal(providerSpec.Value.Raw, config)
 	if err != nil {

